### ISSUE TITLE
praisonai-train: `from-trials --qc` drops 100% of English rows via the Tamil script default and blames the wrong cause; `validate` alone lacks the zero-row guard and atomic write its siblings have

### ISSUE THREAD

**Issue by MervinPraison** [MervinPraison/PraisonAI#4315]
## Summary

The README's own **English-by-construction** workflow silently drops every row and reports
nothing useful. `from-trials --qc` passes no QC config, so `filter_rows(rows, None)` falls back to
the Tamil script range and rejects 100% of English agent trajectories.

```text
$ praisonai-train from-trials trials.json -o train.jsonl --qc     # English trajectories
exit_code = 1
  written=0 skipped_failed=2 skipped_tool_runs=0 skipped_no_text=0
  skipped_unscored=0 skipped_saturated=0
  wrote 0 rows -> train.jsonl
error: no rows exported — check --all/--include-saturated or whether the report has
       passing, tool-free attempts

PC-1  same trials, NO --qc      -> exit 0, written=2, 660 B    (the data is fine)
PC-2  Tamil trials, WITH --qc   -> exit 0, written=1, 567 B    (--qc is not broken per se)
```

Two rows vanish and **no counter accounts for them** — every `skipped_*` is 0 while `written=0` —
and the error text names three possible causes, none of which is the actual one.

This is `README.md:121-156`, *"Verify → export → train → re-verify … fine-tune on the agent's own
verified behaviour"*, with *"Add `--qc` to run rows through the same QC filter as `validate`."*
Agent trajectories are English by construction.

## `validate` has the same default, plus two defects its siblings do not

```text
$ praisonai-train validate english.jsonl --out clean.jsonl
exit_code = 0
  in=5  kept=0 (0.0%)   drops: {'low_script_purity': 5}
  wrote 0 filtered rows -> clean.jsonl
  clean.jsonl: size=0B lines=0

$ praisonai-train validate corpus.jsonl --out corpus.jsonl     # --out aliases the input
exit_code = 0
  in=5  kept=0 (0.0%)
  corpus.jsonl: size=0B lines=0      <-- the corpus is gone
```

1. **`kept=0` exits 0.** Siblings `generate` (`data.py:188-191`) and `from-trials`
   (`data.py:301-304`) both `Exit(1)` on zero rows. `validate` does not.
2. **No zero-row guard and no atomic write.** `validate` (`data.py:356-359`) uses a bare
   `open(out_path, "w")`. `generate` (`data.py:143-184`) and `dedup` (`data.py:234-259`) both use
   `tempfile.mkstemp` + `os.replace` with an explicit *"Only replace the destination when at least
   one row was produced, so the all-failures path leaves any existing file intact"* guard — and
   the helpers (`_apply_replacement_mode`, `_resolve_replace_target`) live **in the same file**.

Control: `dedup corpus2.jsonl --out corpus2.jsonl` → `in=5 kept=5 removed=0`, file intact at
1160 B. The sibling really is safe; the difference is `validate`'s alone.

## What I am NOT claiming — a validator refuted it

An earlier framing of this said *"the Tamil default is undocumented"*. **That is wrong**, and I am
withdrawing it:

- `praisonai_train/setup/validate.yaml:7` ships `script_range: [2944, 3071]  # Tamil block
  U+0B80-U+0BFF; change per language`
- `tests/unit/data/test_data.py:17` deliberately asserts an English output is dropped as
  `low_script_purity`

The default is **intended and pinned by a test**. The real gap is narrower: the README points only
at `setup/config.yaml` (L76) and never at `validate.yaml`, and neither the README nor `--help`
mentions `script_range` at all — while `--help` describes the command generically as *"Quality-check
a dataset (dedup, boilerplate/refusal, script purity, diversity)"*.

## Sweep, proving the range is the sole cause

```text
PC-A  Tamil data, default config       -> in=5 kept=5 (100.0%)  drops: {}
PC-B  English data, script_range: [65, 591] -> in=5 kept=5 (100.0%)  drops: {}
      English data, default config     -> in=5 kept=0 (0.0%)   drops: {'low_script_purity': 5}
```

## Suggested fix

1. Have `from_trials` pass a QC config appropriate to its own output (agent trajectories are the
   caller's language, not Tamil), or omit the script check entirely for that path.
2. Make the zero-row error message name the actual drop reasons — the `drops:` dict is already
   computed and would have said `low_script_purity` immediately.
3. Give `validate` the zero-kept guard and the temp+`os.replace` write its two siblings already
   have, using the helpers in the same module.
4. Document `script_range` and its Tamil default in the README and `--help`.

## Prior art

No issue covers `validate` / script purity / `script_range` / `low_script_purity` (five searches,
`--state all`). #4238 filed the **identical** truncation defect against `generate` — *"The sibling
`dedup` command does it correctly — temp file plus `os.replace`, with a regression test … Apply
that pattern here"* — the fix was applied to `generate`, and **`validate` was left behind**. This is
an uncovered residual, not a re-report.

## Validation record

Two independent refutation attempts. Both reproduced the mechanism; **Validator B partially
refuted the framing** by finding `validate.yaml:7` and the pinning test, which is why the
"undocumented default" claim is withdrawn above and the issue leads with `from-trials --qc`
instead. Validator A supplied the sibling controls showing `dedup` in-place is safe and
`from-trials` exits 1 on zero rows.

## Not covered

`validate` reads the whole input before writing, so `--out` aliasing is not a read-truncation
race — the loss is strictly the zero-kept overwrite.

---

*Audited against `origin/main` @ `773a27bb0`. No network (`ATTEMPTS: []`). `subprocess` globally
stubbed; nothing installed, upgraded or removed.*
